StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace StringCalculator;
+public class StringCalculator
+{
+    private string[] seperators = new string[] { ",", "\\n" };
+    private List<string> negativeNumbers = new List<string>();
+    private int maxNumberToAdd = 1000;
+    private int finalResult = 0;
+
+    public int Add(string numbersGiven)
+    {
+        string numbersToCalculate = numbersGiven;
+
+        if (String.IsNullOrEmpty(numbersToCalculate))
+        {
+            return 0;
+        }
+
+        numbersToCalculate = CheckForSeperators(numbersToCalculate);
+        String[] splitNumbersToAdd = numbersToCalculate.Split(seperators, StringSplitOptions.None);
+
+        CalculateSplitNumberAddition(splitNumbersToAdd);
+
+        return finalResult;
+    }
+
+    private void CalculateSplitNumberAddition(string[] splitNumbersToAdd)
+    {
+        foreach (string currentNumber in splitNumbersToAdd)
+        {
+            int convertedCurrentNumber = Convert.ToInt32(currentNumber);
+
+            if (convertedCurrentNumber < 0)
+            {
+                negativeNumbers.Add(currentNumber);
+                continue;
+            }
+            else if (convertedCurrentNumber > maxNumberToAdd)
+            {
+                continue;
+            }
```

Comment:
Technically this is what we'd refer to as "hidden functionality" - it's something that happens in a method that the method name does not suggest (also could be referred to as a "side effect"). If another developer was using this class and couldn't see its implementation, they'd assume that it does whatever its name suggests - in this case they'd get that, and an exception for negative numbers _and_ filtered out numbers over a 1000 too.

Reply:
This can also be a good indication that your method is dealing with too many responsibilities.